### ISSUE TITLE
how to implement changes in model constants from python

### ISSUE THREAD

**Issue by guidov** [aperezhortal/pySPEEDY#6]
Could you advise on how you might change the model physical constants , during initialization from python?
physical_constants.f90
    ! Physical constants for dynamics
    real(p), parameter :: rearth = 6.371e+6    !! Radius of Earth (m)
    real(p), parameter :: omega  = 7.292e-05   !! Rotation rate of Earth (rad/s)
    real(p), parameter :: grav   = 9.81        !! Gravitational acceleration (m/s/s)

I could see this being changed in the registry, but not sure how it would be done compared with the state variables.

Thanks

**Comment by aperezhortal** [comment]
Hi Guido, 
Unfortunately, those are fixed parameters defined at compile time. That is, they are hardcoded since they correspond to physical constants. 

You can always change the parameters in the F90 code and recompile pySpeedy. But you will need to do this each time you want to change the parameters. 

What is your use case? Why do you need to change these parameters? I am asking because the constant variables in the model can always be converted to dynamic ones,  and expose them to python.

**Comment by guidov** [comment]
Hi, 
It is for an atmospheric dynamics teaching environment. The students know some python, but not Fortran and the model will be preinstalled within a jupyterlab containerized environment in the cloud. I guess they could pull up a terminal and go and edit the sources in the python installation, but they won't be permanent when they leave the session. I guess there could be a script executed from the notebook that does this also but I'm just looking for a way to make things clean, quick and simple, like the way the start and end runtime dates are entered from python. I was planing on changing the solar_radiation module too to add obliquity, precession and eccentricity, and have these exposed to python, but this may be a bit involved. The developmental Julia version of Speedy solves some of these issues, but right now I need to stick with python and this version of Speedy.
Thanks
Guido

**Comment by guidov** [comment]
Hi, 
I tried to setup some of the physical constants as model state variables as in the template but there ends up being a circular conflict in the defining physical constants as state variables in the geometry module. I don't think there is an easy work around.
I tried also setting some dynamic variables in the physical_constants.f90 routine and then using ctypes to access the variables, but I don't know how you would access the shared object in memory that is loaded by the python kernel to access these dynamic variables, or if this is even possible. All this is out of my expertise, so I'm going to recompile everytime there is a change of physical_constants as it doesn't take too long. I'm closing this issue now.
